pkg/network/ebpf/c/http-buffer.h

```diff
@@ -7,10 +7,12 @@
 // transaction object, and returns the number of bytes of the valid data. The number of
 // bytes are used in userspace to zero out the garbage we may have read into the buffer.
 static __always_inline void read_into_buffer(char *buffer, char *data, size_t data_size) {
+    int errno;
     __builtin_memset(buffer, 0, HTTP_BUFFER_SIZE);
 
     // we read HTTP_BUFFER_SIZE-1 bytes to ensure that the string is always null terminated
-    if (bpf_probe_read_user(buffer, HTTP_BUFFER_SIZE - 1, data) < 0) {
+    bpf_probe_read_user_with_telemetry_ret_err(buffer, HTTP_BUFFER_SIZE - 1, data, errno);
```

Comment:
The regular `bpf_probe_read_user_with_telemetry` does not return an error?

Reply:
I could not think of a way to return a value from a macro. So instead this "*_ret_err" macro takes a user defined variable and sets that to error.